### PR TITLE
[Feature] Add `grid-rows-fill`, `grid-rows-fit`, `grid-cols-fill`, and `grid-cols-fit` utilities

### PR DESCRIPTION
**Pull request by jvanderen1**
## Issue

There was no clean way to specify `auto-fill` or `auto-fit` inside a grid template CSS property.

## Background

When #1403, this was not a real feasible solution before the JIT compiler. However, generating these styles on the fly shouldn't have nearly the performance hit as it did before.

Further discussions have been made in #8705

## Changes

I added `grid-rows-fill-*`, `grid-rows-fit-*`, `grid-cols-fill-*`, and `grid-cols-fit-*` to our utilities. The default values are all derived from `theme('spacing')` and arbitrary values will populate correctly with minimum syntax required.

## Example Syntax

| Class | Value |
| ----------- | ----------- |
| `grid-rows-fill-16`      | `grid-template-rows: repeat(auto-fill, minmax(4rem, 1fr))`       |
| `grid-rows-fill-[200px]`      | `grid-template-rows: repeat(auto-fill, minmax(200px, 1fr))`       |
| `grid-cols-fill-20`   | `grid-template-columns: repeat(auto-fill, minmax(5rem, 1fr))`        |
| `grid-cols-fill-[300px]`      | `grid-template-columns: repeat(auto-fill, minmax(300px, 1fr))`       |
| `grid-rows-fit-24`      | `grid-template-rows: repeat(auto-fit, minmax(6rem, 1fr))`       |
| `grid-rows-fit-[400px]`      | `grid-template-rows: repeat(auto-fit, minmax(400px, 1fr))`       |
| `grid-cols-fit-28`   | `grid-template-columns: repeat(auto-fit, minmax(7rem, 1fr))`        |
| `grid-cols-fit-[500px]`   | `grid-template-columns: repeat(auto-fit, minmax(500px, 1fr))`        |

## Remarks

Again, thank you for making a beautiful tool. This is my favorite tool to advocate for ❤️ .

If you have any comments/suggestions, I would love to hear your feedback.

Thanks again.

### CODE REVIEW

File: tests/parallel-variants.test.js [hunk 1]
```diff
@@ -67,12 +67,12 @@ test('parallel variants can be generated using a function that returns parallel
...
-      .test\:font-medium *::test {
-        font-weight: 500;
-      }
       .test\:font-bold::test {
         font-weight: 700;
       }
+      .test\:font-medium *::test {
+        font-weight: 500;
+      }
       .test\:font-medium::test {
```
Comment (jvanderen1) [inline tests/parallel-variants.test.js:76]: I am not sure why, but I had to move the order of 2 classes in 2 separate tests.

**Comment by florianbouvot** [comment]
@jvanderen1 I would love to have this in tailwind 👍

**Comment by lukeocodes** [comment]
This would be absolutely fantastic...

File: tests/prefers-contrast.test.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { run, html, css, defaults } from './util/run'
 it('should be possible to use contrast-more and contrast-less variants', () => {
   let config = {
     content: [
-      { raw: html`<div class="contrast-more:bg-pink-500 contrast-less:bg-black bg-white"></div>` },
+      { raw: html`<div class="bg-white contrast-more:bg-pink-500 contrast-less:bg-black"></div>` },
```
Comment (jvanderen1) [inline tests/prefers-contrast.test.js:6]: Looks like eslint made these changes.

**Comment by jvanderen1** [comment]
Awesome 🥳! I was able to fold:
- `gridTemplateRowsFill` and `gridTemplateRowsFit` into `gridTemplateRows`
- `gridTemplateColsFill` and `gridTemplateColsFit` into `gridTemplateCols`

## Benefits
- Less file updates are needed
- (I believe) Tailwind IntelliSense should work better now too.

**Comment by jvanderen1** [comment]
Hey @adamwathan! I was wondering if you had any thoughts this feature? Is it something the team would be open to adopting?

**Comment by adamwathan** [comment]
repeat(auto-fill, minmax(4rem, 1fr))

**Pull request closed without merging**